src/lib/caching/structures/presences/PresenceGameBuilder.ts

```diff
@@ -0,0 +1,271 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import {
+	ActivityFlags,
+	ActivityType,
+	APIActivityData,
+	APIActivityDataAssets,
+	APIActivityDataEmoji,
+	APIActivityDataParty,
+	APIActivityDataSecrets,
+	APIActivityDataTimestamps
+} from '@klasa/dapi-types';
+
+/**
+ * The presence game builder.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/topics/gateway#activity-object-activity-structure
+ */
+export class PresenceGameBuilder implements APIActivityData {
+
+	/**
+	 * The activity's name.
+	 * @since 0.0.1
+	 */
+	public name: string;
+
+	/**
+	 * The activity's type.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/topics/gateway#activity-object-activity-types
+	 */
+	public type: ActivityType;
+
+	/**
+	 * Stream url, is validated when type is 1.
+	 * @since 0.0.1
+	 */
+	public url?: string | null;
+
+	/**
+	 * Unix timestamp of when the activity was added to the user's session.
+	 * @since 0.0.1
+	 */
+	public created_at: number;
+
+	/**
+	 * Unix timestamps for start and/or end of the game.
+	 * @since 0.0.1
+	 */
+	public timestamps?: APIActivityDataTimestamps[];
```

Comment:
This type doesn't look right

Reply:
Well, those are [your types](https://github.com/dirigeants/dapi-types/blob/c85a5cd561bd7db1d7f8a50a7c998c43d25311c5/src/index.ts#L8-L26), this class just implements it.

Reply:
Oh that type is completely wrong, wtf. I don't remember adding that one myself..brb